src/include/snapper/dialogs.rb

```diff
@@ -47,198 +44,409 @@ def initialize_snapper_dialogs(include_target)
-          # popup label, %{num} is number
-          Label(_("Modify Snapshot %{num}") % { :num => num }),
-          snapshot_term.call("", snapshot)
-        )
-      else
-        cont = VBox(
-          # popup label, %{pre} and %{post} are numbers
-          Label(_("Modify Snapshot %{pre} and %{post}") % { :pre => pre_num, :post => num }),
-          # label
-          Left(Label(_("Pre (%{pre})") % { :pre => pre_num })),
-          snapshot_term.call("pre_", pre_snapshot),
-          VSpacing(),
-          # label
-          Left(Label(_("Post (%{post})") % { :post => num })),
-          snapshot_term.call("", snapshot)
-        )
+      UI.CloseDialog
+      if ret == :ok
+        modified = snapshot.update(args) if snapshot_modified(snapshot, args)
+
+        if snapshot.post? && snapshot_modified(snapshot.pre, pre_args)
+          modified = snapshot.pre.update(pre_args)
+        end
       end
 
-      UI.OpenDialog(
-        Opt(:decorated),
-        HBox(
-          HSpacing(1),
-          VBox(
-            VSpacing(0.5),
-            HSpacing(65),
-            cont,
-            VSpacing(0.5),
-            ButtonBox(
-              PushButton(Id(:ok), Label.OKButton),
-              PushButton(Id(:cancel), Label.CancelButton)
-            ),
-            VSpacing(0.5)
-          ),
-          HSpacing(1)
-        )
-      )
+      modified
+    end
+
+    # Popup for creating new snapshot
+    # @return true if new snapshot was created
+    def CreateSnapshotPopup(pre_snapshots)
+      created = false
+      pre_items = pre_snapshots.map do |s|
+        Item(Id(s), s.to_s)
+      end
+
+      open_create_dialog(pre_items)
+
+      enable_buttons([:post, :pre_list], !pre_items.empty?)
 
       ret = nil
       args = {}
-      pre_args = {}
 
-      while true
+      loop do
         ret = UI.UserInput
         args = {
-          "num"         => num,
-          "description" => UI.QueryWidget(Id("description"), :Value),
-          "cleanup"     => UI.QueryWidget(Id("cleanup"), :Value),
-          "userdata"    => get_userdata("userdata")
+          type:        UI.QueryWidget(Id(:rb_type), :Value),
+          description: UI.QueryWidget(Id("description"), :Value),
+          pre_number:  UI.QueryWidget(Id(:pre_list), :Value),
+          cleanup:     UI.QueryWidget(Id("cleanup"), :Value),
+          user_data:   userdata("userdata")
         }
-        if type == :POST
-          pre_args = {
-            "num"         => pre_num,
-            "description" => UI.QueryWidget(Id("pre_description"), :Value),
-            "cleanup"     => UI.QueryWidget(Id("pre_cleanup"), :Value),
-            "userdata"    => get_userdata("pre_userdata")
-          }
-        end
         break if ret == :ok || ret == :cancel
       end
+
       UI.CloseDialog
-      if ret == :ok
-        if snapshot_modified(snapshot, args)
-          modified = Snapper.ModifySnapshot(args)
+      created = Snapper.CreateSnapshot(args) if ret == :ok
+      created
```

Comment:
In this case, i often have some doubts about readability and comprehension returning from a ternary operator and with an equality comparison but also false is at top and not obvious so :+1: